src/containers/OLDatasetRenderer/OLDatasetRenderer.js

```diff
@@ -0,0 +1,247 @@
+            let h1 = '';
+            text.replace(/<h1>(.*?)<\/h1>/g, (match, g1) => h1 = g1);
+            throw new Error((h1.length)? h1: text)
+          }
+          const index = currentWmsCapabilitiesUrl.indexOf('?');
+          if(index < 0) {
+            throw new Error('Could not find exact WMS Server URL needed for following GetMetadata requests');
+          }
+          const wmsServerUrl = currentWmsCapabilitiesUrl.substring(0, index);
+          // We cannot trust anymore what's returned by GetCapabilities: capabilities['Service']['OnlineResource'];
+
+          /*
+           here we assume that the layer that actually contains the information we want to display is the first one of the dataset
+           for instance, there could be layers containing lat or lon in the return values of the capabilities
+           hopefully the relevant variable will always be the first one
+           */
+          const layer = capabilities['Capability']['Layer']['Layer'][0]['Layer'][0];
+          const layerName = layer['Name'];
+          const minMaxBracket = `${dataset['variable_min']},${dataset['variable_max']}`;
+          console.log('current dataset min max bracket: %s', minMaxBracket);
+          this.createDatasetLayer(minMaxBracket, layerName, wmsServerUrl, dataset.opacity, `default-scalar/${dataset.variable_palette}`);
+
+          if (layer['Dimension']) {
+            const timeDimension = this.findDimension(layer['Dimension'], 'time');
+            const date = timeDimension.values.substring(0, 24);
+            this.props.layerDatasetActions.fetchWMSLayerTimesteps(wmsServerUrl, layerName, date);
+            // this.props.layerDatasetActions.setCurrentDateTime(date);
+          }
+
+          this.props.layerDatasetActions.setSelectedDatasetCapabilities(capabilities);
+          this.props.layerDatasetActions.fetchWMSLayerDetails(wmsServerUrl, layerName);
+          // Normally fetched entirely by OpenLayers, but we want to an error when returning an error: ex. 401 Unauthorized
+          this.props.layerDatasetActions.testWMSGetMapPermission(wmsServerUrl, layerName);
+        })
+        .catch(err => {
+          console.log(err);
+          NotificationManager.error(`Method GetCapabilities failed at being fetched from the NcWMS2 server: ${err}`, 'Error', 10000);
+        });
+    }
+  }
+
+  updateColorPalette (palette) {
+    if (this.layer) {
+      this.source.updateParams({
+        'STYLES': `default-scalar/${palette}`
+      });
+    } else {
+      // NotificationManager.warning('Please display a dataset before changing the color palette.', 'Warning', 10000);
+    }
+  }
+
+  findDimension (dimensions, dimensionName) {
+    for (let i = 0; i < dimensions.length; i++) {
+      if (dimensions[i]['name'] === dimensionName) {
+        return dimensions[i];
+      }
+    }
+  }
+
+  datasetHasWmsUrls (dataset) {
+    return !!(dataset.wms_url && dataset.wms_url.length > 0);
```

Comment:
ahh, the good old Bang! Bang! You're Boolean™ operator :)

Reply:
Ça c'est ton code, je n'ai littéralement jamais eu besoin d'écrire une double inversion.

Reply:
C'est clair! Je ne te l'attribuais / ne chiâlais pas :)

Reply:
Dans, mon cas je trouve ça vraiment horrible !

Reply:
D'accord, veux-tu le modifier ou que je le fasse post merge? Ajouter un TODO?

Reply:
Concrètement il s'agit de faire le cast vers boolean de façon explicite plutôt qu'implicite ailleurs dans le code.